domain/resource/state/resource.go

```diff
@@ -1615,6 +1615,179 @@ func (st *State) buildResourcesToAdd(
 	return resources, result, nil
 }
 
+// UpdateUploadResourceAndDeletePriorVersion deletes a reference to the old
+// stored blob, saving the hash to return. Adds a new row in the resource
+// table which indicates the resource will be updated. Next, it sets it on the
+// application_resource table, removing the old resource for this charm
+// resource.
+//
+// The following error types can be expected to be returned:
+//   - [resourceerrors.ResourceNotFound] is returned if the resource cannot be
+//     found.
+func (st *State) UpdateUploadResourceAndDeletePriorVersion(
```

Comment:
Is the plan still to delete the old blobs here? I thought we were leaving that for the cleanup worker now.

Reply:
Deleting the old blob matches the behavior of UpdateResourceRevisionAndDeletePriorVersion. It's also in the description of changes. 

The resource row will be left the a cleanup worker to remove at this point.